tests/search.service.spec.js

```diff
@@ -61,4 +70,43 @@ describe("Searching for hackers", function() {
                 done();
             });
     })
+    it("Should only return 1 hacker (page size)", function(done) {
+        chai.request(server.app)
+            .get("/api/search/hacker")
+            .query({q: JSON.stringify(query2), limit:1})
+            .end(function(err, res) {
+                res.should.have.status(200);
+                res.body.data.should.have.length(1);
+                done();
+            })
+    })
+    it("Should only return 1 hacker (pagination)", function(done) {
+        chai.request(server.app)
+            .get("/api/search/hacker")
+            //There are two test samples so by making limit 1, there will be something on the second page
+            .query({q: JSON.stringify(query2), limit:1, page: 1})
+            .end(function(err, res) {
+                res.should.have.status(200);
+                res.body.data.should.have.length(1);
+                done();
+            })
+    })
+    it("Should throw an error because out of bounds (page size)" , function(done) {
+        chai.request(server.app)
+            .get("/api/search/hacker")
+            .query({q: JSON.stringify(query2), limit:5000})
```

Comment:
Does this throw an error because there is no `page` parameter here?

Reply:
Nah, because trying to retrieve more than 1000 queries at one, I set a max so they couldn't retrieve they entire results through whatever is going to be calling this